packages/rlc-common/src/metadata/packageJson/azurePackageCommon.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  PackageCommonInfoConfig,
+  getCommonPackageDevDependencies
+} from "./packageCommon.js";
+
+export interface AzurePackageInfoConfig extends PackageCommonInfoConfig {
+  hasLro: boolean;
+  hasPaging: boolean;
+  specSource: "Swagger" | "TypeSpec";
+}
+
+/**
+ * Build the common package.json config for an Azure package.
+ */
+export function getAzureCommonPackageInfo(config: AzurePackageInfoConfig) {
+  return {
+    keywords: ["node", "azure", "cloud", "typescript", "browser", "isomorphic"],
+    author: "Microsoft Corporation",
+    license: "MIT",
+    ...getAzureCjsCommonInfo(config),
+    ...getAzureEsmCommonInfo(config)
+  };
+}
+
+/**
+ * Builds the common dependencies for an Azure package.
+ */
+export function getAzurePackageDependencies({
+  hasLro,
+  hasPaging
+}: AzurePackageInfoConfig) {
+  let dependencies: Record<string, string> = {
+    "@azure-rest/core-client": "^1.2.0",
+    "@azure/core-auth": "^1.6.0",
+    "@azure/core-rest-pipeline": "^1.5.0",
+    "@azure/logger": "^1.0.0",
+    tslib: "^2.6.2"
+  };
+
+  if (hasLro) {
+    dependencies = {
+      ...dependencies,
+      "@azure/core-lro": "^2.5.4",
+      "@azure/abort-controller": "^1.0.0"
+    };
+  }
+
+  if (hasPaging) {
+    dependencies = {
+      ...dependencies,
+      "@azure/core-paging": "^1.5.0"
+    };
+  }
+
+  return dependencies;
+}
+
+function getAzureCjsCommonInfo({
+  withTests,
+  name,
+  nameWithoutScope,
+  moduleKind
+}: AzurePackageInfoConfig) {
+  if (moduleKind !== "cjs") {
+    return {};
+  }
+
+  return {
+    files: [
+      "dist/",
+      withTests ? "dist-esm/src/" : "dist-esm/",
```

Comment:
I think we should consider a case where generateSample: true and generateTest: false?

Reply:
you mean (withTests || withSamples) ?   "dist-esm/src/" : "dist-esm/",

Reply:
yeah, I found this bug in my pr https://github.com/Azure/autorest.typescript/pull/2342